src/axis.js

```diff
@@ -1,16 +1,386 @@
+                    textUpdate.attr("x", $$$.tickLength)
+                        .attr("y", 0)
+                        .style("text-anchor", "start");
+                    tspanUpdate.attr('x', $$$.tickLength)
+                        .attr("dy", function (d, i) { return $$$.tspanDy(d, i); });
+                    pathUpdate.attr("d", "M" + $$$.outerTickSize + "," + $$$.range[0] + "H0V" + $$$.range[1] + "H" + $$$.outerTickSize);
+                    break;
+                }
+            }
+            if (scale1.rangeBand) {
+                var x = scale1, dx = x.rangeBand() / 2;
+                scale0 = scale1 = function (d) {
+                    return x(d) + dx;
+                };
+            } else if (scale0.rangeBand) {
+                scale0 = scale1;
+            } else {
+                tickExit.call(tickTransform, scale1, $$$.tickOffset);
+            }
+            tickEnter.call(tickTransform, scale0, $$$.tickOffset);
+            tickUpdate.call(tickTransform, scale1, $$$.tickOffset);
+        });
+    }
+    axis.scale = function (x) {
+        if (!arguments.length) { return $$$.scale; }
+        $$$.scale = x;
+        return axis;
+    };
+    axis.orient = function (x) {
+        if (!arguments.length) { return $$$.orient; }
+        $$$.orient = x in {top: 1, right: 1, bottom: 1, left: 1} ? x + "" : "bottom";
+        return axis;
+    };
+    axis.tickFormat = function (format) {
+        if (!arguments.length) { return $$$.tickFormat; }
+        $$$.tickFormat = format;
+        return axis;
+    };
+    axis.tickCentered = function (isCentered) {
+        if (!arguments.length) { return $$$.tickCentered; }
+        $$$.tickCentered = isCentered;
+        return axis;
+    };
+    axis.tickOffset = function () {
+        return $$$.tickOffset;
+    };
+    axis.tickInterval = function () {
+        var interval, length;
+        if (params.isCategory) {
+            interval = $$$.tickOffset * 2;
+        }
+        else {
+            length = axis.g.select('path.domain').node().getTotalLength() - $$$.outerTickSize * 2;
+            interval = length / axis.g.selectAll('line').size();
+        }
+        return interval === Infinity ? 0 : interval;
+    };
+    axis.ticks = function () {
+        if (!arguments.length) { return $$$.tickArguments; }
+        $$$.tickArguments = arguments;
+        return axis;
+    };
+    axis.tickCulling = function (culling) {
+        if (!arguments.length) { return $$$.tickCulling; }
+        $$$.tickCulling = culling;
+        return axis;
+    };
+    axis.tickValues = function (x) {
+        if (typeof x === 'function') {
+            $$$.tickValues = function () {
+                return x($$$.scale.domain());
+            };
+        }
+        else {
+            if (!arguments.length) { return $$$.tickValues; }
+            $$$.tickValues = x;
+        }
+        return axis;
+    };
+    return axis;
+};
+
+export default function Axis(owner) {
```

Comment:
How about using class keyword like the below?

```js
export default class Axis extends Component {
  constructor (owner) {
    ...
  }
}
```

By the above we don't need to maintain `inherit` util anymore.